libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1948,3 +1948,32 @@ void TypeChecker::requireLValue(Expression const& _expression)
 		m_errorReporter.typeError(_expression.location(), "Expression has to be an lvalue.");
 }
 
+void TypeChecker::overrideTypeError(FunctionDefinition const& function, FunctionDefinition const& super)
+{
+	string message;
+
+	if (function.visibility() != super.visibility())
+		message = "Overriding function visibility differs from " + super.fullyQualifiedName() + ".";
+	else if (function.isDeclaredConst() && !super.isDeclaredConst())
+		message = "Overriding function should not be declared constant.";
+	else if (!function.isDeclaredConst() && super.isDeclaredConst())
+		message = "Overriding function should be declared constant.";
+	else if (function.isPayable() && !super.isPayable())
+		message = "Overriding function should not be declared payable.";
+	else if (!function.isPayable() && super.isPayable())
+		message = "Overriding function should be declared payable.";
+
+	if (message.empty())
+	{
+		FunctionType functionType(function);
+		FunctionType superType(super);
+
+		if (functionType != superType)
+			message = "Overriding function return types differ from " + super.fullyQualifiedName() + ".";
+	}
+
+	if (message.empty())
```

Comment:
None at the moment. It's just a fallback in case conditions change in the caller. Now that I think about it, perhaps it's better to do a `solAssert(!message.empty(), ...)`?